aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -51,9 +53,13 @@ __global__ void kernelHistogram1D(
       // Convert `linearIndex` into an offset of `b`
       const IndexType bOffset =
           detail::IndexToOffset<input_t, IndexType, BDims>::get(linearIndex, b);
-      // Use value at `b` as an offset of `smem`
-      const IndexType pOffset = b.data[bOffset] / binsize;
-      atomicAdd(&smem[pOffset], getOp(linearIndex));
+      const auto bVal = b.data[bOffset];
+      if (bVal >= minvalue && bVal <= maxvalue) {
+        // Use value at `b` as an offset of `smem`
+        IndexType bin = (int)((bVal - minvalue) * nbins / (maxvalue - minvalue));
```

Comment:
Can you refactor this into its own function? It comes up three times and although it is short it is nice to know that if we change this in the future, we only need to touch one location